src/encoded/batch_download.py

```diff
@@ -546,147 +313,6 @@ def peak_metadata(context, request):
-    rows = []
-    for experiment_json in results['@graph']:
-        if experiment_json.get('files', []):
-            exp_data_row = []
-            for column in header:
-                if not _tsv_mapping[column][0].startswith('files'):
-                    make_cell(column, experiment_json, exp_data_row)
-
-            f_attributes = ['files.title', 'files.file_type', 'files.file_format',
-                            'files.file_format_type', 'files.output_type', 'files.assembly']
-
-            for f in experiment_json['files']:
-                if not files_prop_param_list(f, param_list):
-                    continue
-                if visualizable_only and not is_file_visualizable(f):
-                    continue
-                if raw_only and f.get('assembly'):
-                    # "raw" option only allows files w/o assembly.
-                    continue
-                if restricted_files_present(f):
-                    continue
-                if is_no_file_available(f):
-                    continue
-                f['href'] = request.host_url + f['href']
-                f_row = []
-                for attr in f_attributes:
-                    f_row.append(f.get(attr[6:], ''))
-                data_row = f_row + exp_data_row
-                for prop in file_attributes:
-                    if prop in f_attributes:
-                        continue
-                    path = prop[6:]
-                    temp = []
-                    for value in simple_path_ids(f, path):
-                        temp.append(str(value))
-                    if prop == 'files.replicate.rbns_protein_concentration':
-                        if 'replicate' in f and 'rbns_protein_concentration_units' in f['replicate']:
-                            temp[0] = temp[0] + ' ' + f['replicate']['rbns_protein_concentration_units']
-                    if prop in ['files.paired_with', 'files.derived_from']:
-                        # chopping of path to just accession
-                        if len(temp):
-                            new_values = [t[7:-1] for t in temp]
-                            temp = new_values
-                    data = list(set(temp))
-                    data.sort()
-                    data_row.append(', '.join(data))
-                audit_info = [make_audit_cell(audit_type, experiment_json, f) for audit_type in _audit_mapping]
-                data_row.extend(audit_info)
-                rows.append(data_row)
-    fout = io.StringIO()
-    writer = csv.writer(fout, delimiter='\t', lineterminator='\n')
-    header.extend([prop for prop in _audit_mapping])
-    writer.writerow(header)
-    writer.writerows(rows)
-    return Response(
-        content_type='text/tsv',
-        body=fout.getvalue(),
-        content_disposition='attachment;filename="%s"' % 'metadata.tsv'
-    )
-
-
 @view_config(route_name='batch_download', request_method=('GET', 'POST'))
 def batch_download(context, request):
```

Comment:
batdh_download and metadata_tsv do a lot of similar things, but not the _same_ things. But I had wondered if they could share some code, but it does seem kind of difficult to do, when the shared code might be a line or two here or there.

Reply:
I think we need to hoist batch_download endpoints over next. The file filtering stuff that was fixed for metadata will still be broken here for example.